dotnet/src/SemanticKernel.Core/Functions/OrderedIAIServiceSelector.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.SemanticKernel.AI;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Services;
+
+namespace Microsoft.SemanticKernel.Functions;
+
+/// <summary>
+/// Implementation of <see cref="IAIServiceSelector"/> that selects the AI service based on the order of the model settings.
+/// Uses the service id to select the preferred service provider and then returns the service and associated model settings.
+/// </summary>
+internal class OrderedIAIServiceSelector : IAIServiceSelector
+{
+    /// <inheritdoc/>
+    public (T?, AIRequestSettings?) SelectAIService<T>(IAIServiceProvider serviceProvider, IReadOnlyList<AIRequestSettings>? modelSettings) where T : IAIService
+    {
+        if (modelSettings == null || modelSettings.Count == 0)
+        {
+            var service = serviceProvider.GetService<T>(null);
+            if (service != null)
+            {
+                return (service, null);
+            }
+        }
+        else
+        {
+            AIRequestSettings? defaultRequestSettings = null;
+            foreach (var model in modelSettings)
+            {
+                if (!string.IsNullOrEmpty(model.ServiceId))
+                {
+                    var service = serviceProvider.GetService<T>(model.ServiceId);
+                    if (service != null)
+                    {
+                        return (service, model);
+                    }
+                }
+                else
+                {
+                    // First request settings with empty or null service id is the default
+                    defaultRequestSettings ??= model;
+                }
+            }
+
+            if (defaultRequestSettings is not null)
+            {
+                return (serviceProvider.GetService<T>(null), defaultRequestSettings);
```

Comment:
Do we allow the `SelectAIService` to return null from this `GetService<T>` call?

The other scenarios we throw, thinking about consistency suggest throwing in here as well.